### ISSUE TITLE
Crash when running on Android 11 (API 30) - `package does not belong to uid`

### ISSUE THREAD

**Issue by bartekpacia** [leancodepl/patrol#1455]
Patrol tests crash before running on Android emulators with API level 30:

```
ActivityTaskManager: Permission Denial: package=pl.leancode.patrol.example.test does not belong to uid=10178
```

<details><summary>More logs</summary>
<p>

```
07-03 14:15:17.045 10859 10859 I PatrolServer: --------------------------------
07-03 14:15:17.045 10859 10859 I PatrolServer: PatrolJUnitRunner.onCreate() (initial run)
07-03 14:15:17.046 10859 10859 D ConnectionBase: Connected to androidx.test.orchestrator.OrchestratorService
07-03 14:15:17.046 10859 10881 D AndroidJUnitRunner: onStart is called.
07-03 14:15:17.046 10859 10859 I MonitoringInstr: No JSBridge.
07-03 14:15:17.049 10859 10881 D TestRequestBuilder: Using class path scanning to discover tests
07-03 14:15:17.049 10859 10881 I TestRequestBuilder: Scanning classpath to find tests in paths [/data/app/~~yDlhYN_cl5-5ePDj_usOBw==/pl.leancode.patrol.example.test-OVzogQeEvdvcF8m-n3Tw-g==/base.apk]
07-03 14:15:17.049 10859 10881 W .patrol.exampl: Opening an oat file without a class loader. Are you using the deprecated DexFile APIs?
07-03 14:15:17.051 10859 10881 I PatrolServer: PatrolJUnitRunner.setUp(): activityClass = pl.leancode.patrol.example.MainActivity
07-03 14:15:17.052   516  2891 W ActivityTaskManager: Permission Denial: package=pl.leancode.patrol.example.test does not belong to uid=10178
07-03 14:15:17.053 10859 10881 D AndroidJUnitRunner: Use the raw file system for managing file I/O.
07-03 14:15:17.054 10859 10881 D TestExecutor: Adding listener androidx.test.internal.runner.listener.LogRunListener
07-03 14:15:17.054 10859 10881 D TestExecutor: Adding listener androidx.test.internal.events.client.TestDiscoveryListener
07-03 14:15:17.054 10859 10881 D TestExecutor: Adding listener androidx.test.internal.runner.listener.TraceRunListener
07-03 14:15:17.055 10859 10881 I TestRunner: run started: 1 tests
07-03 14:15:17.056 10859 10881 I TestRunner: started: initializationError(pl.leancode.patrol.example.MainActivityTest)
07-03 14:15:17.056 10859 10881 E TestRunner: failed: initializationError(pl.leancode.patrol.example.MainActivityTest)
07-03 14:15:17.056 10859 10881 E TestRunner: ----- begin exception -----
07-03 14:15:17.056 10859 10881 E TestRunner: java.lang.SecurityException: Permission Denial: package=pl.leancode.patrol.example.test does not belong to uid=10178
07-03 14:15:17.056 10859 10881 E TestRunner: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:2373)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at android.os.Parcel.createException(Parcel.java:2357)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at android.os.Parcel.readException(Parcel.java:2340)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at android.os.Parcel.readException(Parcel.java:2282)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:3696)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1723)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at androidx.test.runner.MonitoringInstrumentation.execStartActivity(MonitoringInstrumentation.java:605)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at android.app.ContextImpl.startActivity(ContextImpl.java:1023)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at android.app.ContextImpl.startActivity(ContextImpl.java:994)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at pl.leancode.patrol.PatrolJUnitRunner.setUp(PatrolJUnitRunner.java:69)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at pl.leancode.patrol.example.MainActivityTest.testCases(MainActivityTest.java:15)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at java.lang.reflect.Method.invoke(Native Method)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.runners.Parameterized$RunnersFactory.allParameters(Parameterized.java:424)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.runners.Parameterized$RunnersFactory.<init>(Parameterized.java:375)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.runners.Parameterized$RunnersFactory.<init>(Parameterized.java:360)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.runners.Parameterized.<init>(Parameterized.java:303)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at java.lang.reflect.Constructor.newInstance0(Native Method)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at androidx.test.internal.runner.junit4.AndroidAnnotatedBuilder.runnerForClass(AndroidAnnotatedBuilder.java:63)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:37)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at androidx.test.internal.runner.AndroidRunnerBuilder.runnerForClass(AndroidRunnerBuilder.java:149)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at androidx.test.internal.runner.AndroidLogOnlyBuilder.runnerForClass(AndroidLogOnlyBuilder.java:93)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at androidx.test.internal.runner.ScanningTestLoader.doCreateRunner(ScanningTestLoader.java:50)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at androidx.test.internal.runner.TestLoader.getRunnersFor(TestLoader.java:64)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at androidx.test.internal.runner.TestRequestBuilder.build(TestRequestBuilder.java:835)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at androidx.test.runner.AndroidJUnitRunner.buildRequest(AndroidJUnitRunner.java:650)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:418)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2205)
07-03 14:15:17.056 10859 10881 E TestRunner: Caused by: android.os.RemoteException: Remote stack trace:
07-03 14:15:17.056 10859 10881 E TestRunner: 	at com.android.server.wm.ActivityTaskManagerService.assertPackageMatchesCallingUid(ActivityTaskManagerService.java:2598)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at com.android.server.wm.ActivityTaskManagerService.startActivityAsUser(ActivityTaskManagerService.java:1081)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at com.android.server.wm.ActivityTaskManagerService.startActivityAsUser(ActivityTaskManagerService.java:1072)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at com.android.server.wm.ActivityTaskManagerService.startActivity(ActivityTaskManagerService.java:1047)
07-03 14:15:17.056 10859 10881 E TestRunner: 	at android.app.IActivityTaskManager$Stub.onTransact(IActivityTaskManager.java:1422)
07-03 14:15:17.056 10859 10881 E TestRunner:
07-03 14:15:17.056 10859 10881 E TestRunner: ----- end exception -----
07-03 14:15:17.058 10859 10881 I TestRunner: finished: initializationError(pl.leancode.patrol.example.MainActivityTest)
07-03 14:15:17.058 10859 10881 D TestDiscoveryListener: JUnit reported pl.leancode.patrol.example.MainActivityTest#initializationError; discarding as bogus.
07-03 14:15:17.058 10859 10881 I TestRunner: run finished: 1 tests, 1 failed, 0 ignored
07-03 14:15:17.059   516  2891 I ActivityManager: Force stopping pl.leancode.patrol.example appid=10178 user=0: finished inst
07-03 14:15:17.059   516  2891 I ActivityManager: Killing 10859:pl.leancode.patrol.example/u0a178 (adj 0): stop pl.leancode.patrol.example due to finished inst
07-03 14:15:17.059 10841 10841 D AndroidRuntime: Shutting down VM
```

</p>
</details> 

This problem was first discovered by @jakubfijalkowski when running example app's tests on BrowserStack. Today I tried running the same tests on emulator.wtf (emu config: `device=Pixel2,version=30,gpu=auto`) and I'm getting the same problem ([link to run](https://emulator.wtf/o/baad9a1c-4c73-4c4d-9ad8-31c3eed217eb/r/bb9bb84b-e473-4c9a-8884-fb4a33b68a94/j/eb500a40-9c77-428d-ab06-5a341f36a014) - only for members of LeanCode workspace).

I can also reproduce this problem on Android emulator with API 30 (with Google APIs, FWIW).

I made some searching.
- [`ActivityTaskManagerService#assertPackageMatchesCallingUid`](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/services/core/java/com/android/server/wm/ActivityTaskManagerService.java;l=2238;drc=f0216079d06e10e0b45d0a1d7fc53d6f6f55fe71) - the method in AOSP where the exception originates from
- [search for "does not belong to uid" in AOSP codebase](https://cs.android.com/search?q=%22does%20not%20belong%20to%20uid%22)

What's strange/interesting is that running non-Patrol tests with ATO does work on Android 11 emulators - to reproduce this behavior, clone [the official AndroidTestOrchestratorSample](https://github.com/android/testing-samples/tree/main/runner/AndroidTestOrchestratorSample) and `./gradlew :app:connectedDebugAndroidTest`.

Similar issues, might or might not be related:
- [ConnectivityManager registering callback throws security exception ](https://issuetracker.google.com/issues/175055271?pli=1)
- [Need `--force-queryable` when manually installing android test orchestor on API 30 ](https://issuetracker.google.com/issues/262661481)
- [Android instrumentation test fail on Android 11 device](https://github.com/android/android-test/issues/920)
- [Instrumented tests don't work on API 30 using Gradle command](https://github.com/android/android-test/issues/986)

**Comment by gauravmehta13** [comment]
Facing same issue
java.lang.SecurityException: Permission Denial: package=com.health.caremonitor.test does not belong to uid=10274

**Comment by Gaurav-CareMonitor** [comment]
Facing same issue while running app on firebase test labs
java.lang.SecurityException: Permission Denial: package=com.health.caremonitor.test does not belong to uid=10274

**Comment by IvanHorchakov** [comment]
Same [here](https://github.com/leancodepl/patrol/discussions/1479), seems like using API 30 it is not the case now. On API 31 everything is Ok.

**Comment by bartekpacia** [comment]
If you read the links provided, it looks like a bug with Android 11 itself - and there's nothing that can be done about it. Firebase Test Lab, emulator.wtf - everyone has the same problem.

**Comment by jBorkowska** [comment]
Closing, as it is fixed in `patrol 2.2.0`
